ziti/ziti.go

```diff
@@ -569,6 +594,75 @@ func (context *contextImpl) DialWithOptions(serviceName string, options *DialOpt
 
 	return nil, errors.Wrapf(err, "unable to dial service '%s'", serviceName)
 }
+func (context *contextImpl) GetServiceForAddr(network, hostname string, port uint16) (*edge.Service, int, error) {
+	var service *edge.Service
+	score := math.MaxInt
+	context.intercepts.Range(func(key, value any) bool {
+		intercept := value.(*edge.InterceptV1Config)
+		fmt.Println(intercept)
+		sc := intercept.Match(network, hostname, port)
+		if sc != -1 {
+			if score > sc {
+				score = sc
+				service = intercept.Service
+			}
+
+			if sc == 0 {
+				return false
+			}
```

Comment:
So if there are two intercepts with the same score, we return the first one we happen across. Not sure how often that will happen, but is it deterministic?

Reply:
added determinism by using service name -- alphabetically first service wins